### ISSUE TITLE
No compatible build found

### ISSUE THREAD

**Issue by Orangetronic** [fourseven/meteor-scss#60]
The build on atmosphere seems to be missing / broken. When I try to update (or do a clean install) I'm seeing:

    $ meteor update
    This project is already at Meteor 1.0.2.1, the latest release.
    => Errors while upgrading packages:           
                                              
    While checking for fourseven:scss@1.2.3:
    error: No compatible build found

I don't know if this is an atmosphere issue or a meteor-scss issue — I'm fairly new to the world of MeteorJS packages.

**Comment by JeremySaks** [comment]
I'm getting this also.

**Comment by fourseven** [comment]
See #59 - tl;dr is that I did a `1.2.3` release last night but node-sass won't build on linux with that version, so I yanked it about an hour ago. Use `0.9.6`, `1.0.0`, or `2.0.0_beta_1`.

**Comment by Orangetronic** [comment]
:)

About an hr ago I was also seeing:

    $ meteor add fourseven:scss@1.0.0
     => Errors while adding packages:             
                                              
    While checking for fourseven:scss@1.2.3:
    error: No compatible build found

Just tried again and it is now working — I guess atmosphere takes a minute to catch up.

Thanks for sharing a great library! looks like exciting things to come in 2.0.0

**Comment by wfleming** [comment]
I'm still seeing the `error: No compatible build found` message in a lot of cases: I ran `meteor add fourseven:scss@1.0.0` yesterday & today & continue to get that message. I finally just edited `.meteor/versions` & `.meteor/packages` by hand, and that seemed to do the trick, but today I upgraded to meteor 10.3.1, and at the end of the upgrade it logged this message:

```
updated to Meteor 1.0.3.1.
=> Errors while upgrading packages:           
                                              
While checking for fourseven:scss@1.2.3:
error: No compatible build found
```

When this was run, there was no mention of `1.2.3` in my package versions, only `1.0.0`.

Does anybody know if this is a local atmosphere caching issue or something else? I'd already tried wiping out `.meteor/local` and `~/.meteor` (and reinstalling meteor afterwards), so if it's a local cache I'm not sure where it is. It's not exactly broken, but it anybody knows where else on my system atmosphere might be caching info about the `1.2.3` build, I'd appreciate it.

**Comment by JeremySaks** [comment]
@wfleming I am still receiving the error as well. But if you want to run the package at 1.0.0 you need to run `meteor add fourseven:scss@=1.0.0` (note the equals sign). This is described in the docs http://docs.meteor.com/#/full/meteoradd and worked for me to run it at 1.0.0.

Still waiting, however, for a fix or updated release to 1.2.3.

**Comment by wfleming** [comment]
Cheers, @JeremySaks, I missed that about the `=`.

**Comment by fourseven** [comment]
There will never be a 1.2.3 release, the node-sass build chain doesn't compile on the meteor linux package build boxes. I've opened an issue to enquire about yanking that version here - https://github.com/meteor/meteor/issues/3524 and once I get a response I'll probably just publish a 2.0.0-beta_2 to reset the atmosphere preferred version.

**Comment by Orangetronic** [comment]
👍

**Comment by fourseven** [comment]
Use `meteor add fourseven:scss@=0.9.6` or `meteor add fourseven:scss@=2.0.0-beta_1`, 1.0.0 has issues with media queries.

**Comment by mayognaise** [comment]
:+1:

**Comment by tdmoneybanks** [comment]
meteor add foursevel:scss@=0.9.6 worked for me thanks!

**Comment by dcsan** [comment]
anything with an equals sign in seems to hang indefinitely.

```
meteor-app> meteor add fourseven:scss@=2.0.0-beta_1
   
^C
                                             
meteor-app> meteor add fourseven:scss@1.0.0        
Currently using fourseven:scss without any version constraint.
The version constraint will be changed to 1.0.0.
                                              
fourseven:scss: Style with attitude and autoprefixer.
meteor-app> meteor update
This project is already at Meteor 1.0.3.1, the latest release.
=> Errors while upgrading packages:           
                                              
While checking for fourseven:scss@1.2.3:
error: No compatible build found

meteor-app> meteor add fourseven:scss@=0.9.6
   Selecting package versions      

^C
```

**Comment by fourseven** [comment]
What's your environment @dcsan ? Linux x64?

**Comment by dcsan** [comment]
nope, OSX. two machines i know of fail on this.

**Comment by fourseven** [comment]
Alright 1.2.3 hadn't been published for OS X, but now has. 2.0.0-beta_1 should exist for OS X already, I'm not sure why it's hanging. Can you try `SKIP_NODE_SASS_TESTS=true meteor add fourseven:scss@=2.0.0-beta_1` please?

**Comment by fourseven** [comment]
Also, which version of OS X, @dcsan?

**Comment by dcsan** [comment]
OS 10.10.2 (14C109) 
probably same as everyone else given apple's updates

> SKIP_NODE_SASS_TESTS=true meteor add fourseven:scss@=2.0.0-beta_1
   Selecting package versions


hangs.

**Comment by dcsan** [comment]
```
meteor show --show-all fourseven:scss
Package: fourseven:scss@1.2.3                 
Maintainers: fourseven                        
Git: https://github.com/fourseven/meteor-scss.git
                                              
Style with attitude and autoprefixer.         
                                              
Versions:                                     
  0.9.4         August 26th, 2014             
  0.9.5         August 27th, 2014
  0.9.6         October 7th, 2014
  1.0.0         October 17th, 2014    installed
  1.1.4         February 7th, 2015
  1.2.3         January 18th, 2015
  2.0.0-beta    January 7th, 2015
  2.0.0-beta_1  January 15th, 2015
  2.0.0-beta_2  January 27th, 2015
  2.0.0-beta_3  February 6th, 2015

meteor-app> meteor add fourseven:scss@2.0.0-beta_3
   Selecting package versions                |

```

the versions are there, just not working.

**Comment by fourseven** [comment]
I understand your frustration @dcsan - does it work if you `meteor create some-new-project` and then add in both ionic and scss? 

`meteor show` doesn't let you know which packages need binaries vs which ones have been published, which is really a shortcoming (it also won't let you yank a package even as a maintainer).

**Comment by dcsan** [comment]
thanks for all your hard work trying to get this going. it's quite a burden you're taking on.

quick question - if i `meteor deploy` an app that i have successfully got to run locally, are the final compiled CSS files all that's needed? otherwise all these issues with the scss compiler will also manifest at deploy time?

I see you're using libsass. at least that removes the ruby dep but this is why you have to compile the builds for each and every OS yourself?

**Comment by fourseven** [comment]
Both `meteor deploy` and `meteor build` will locally pre-process to css before the bundle is created, and only the output css exists in that bundle.

Early on when I started using Meteor I noticed the lack of Sass support, and for *.js projects (I'm a Ruby developer by day) I dislike requiring Ruby/Compass when libsass/node-sass exists. There's still some rough edges with node-sass and libsass, but I'm hoping to hide most of them with this package. Yes it has native dependancies, which should be downloaded by the meteor package system and not compiled locally (generally).

As I said in the google group, it's the only css pre-processor to not be written in Javascript, so it unfortunately has more moving parts than Stylus or Less. 

I can get `meteor add fourseven:scss@=2.0.0-beta_3` working on my projects, so there could be something else breaking your builds (I'm on OS X 10.10.3 beta for reference). 

Any luck with a fresh project?
